Projects/ChangeTier/Tests/ViewModelTests/ChangeTierViewModelTests.swift

```diff
@@ -0,0 +1,214 @@
+import XCTest
+import hCore
+
+@testable import ChangeTier
+@testable import hCoreUI
+
+final class ChangeTierViewModelTests: XCTestCase {
+    weak var sut: MockChangeTierService?
+
+    let tiers: [Tier] = [
+        .init(
+            id: "id1",
+            name: "standard",
+            level: 1,
+            deductibles: [],
+            premium: .init(amount: "229", currency: ""),
+            displayItems: [],
+            exposureName: "exposureName",
+            productVariant: .init(
+                termsVersion: "",
+                typeOfContract: "",
+                partner: nil,
+                perils: [],
+                insurableLimits: [],
+                documents: [],
+                displayName: "",
+                displayNameTier: nil,
+                displayNameTierLong: nil
+            ),
+            FAQs: []
+        ),
+        .init(
+            id: "id2",
+            name: "max",
+            level: 2,
+            deductibles: [
+                .init(
+                    deductibleAmount: nil,
+                    deductiblePercentage: nil,
+                    subTitle: nil,
+                    premium: .init(amount: "229", currency: "SEK")
+                )
+            ],
+            premium: .init(amount: "229", currency: ""),
+            displayItems: [],
+            exposureName: "exposureName",
+            productVariant: .init(
+                termsVersion: "",
+                typeOfContract: "",
+                partner: nil,
+                perils: [],
+                insurableLimits: [],
+                documents: [],
+                displayName: "",
+                displayNameTier: nil,
+                displayNameTierLong: nil
+            ),
+            FAQs: []
+        ),
+    ]
+
+    override func setUp() {
+        super.setUp()
+        sut = nil
+    }
+
+    override func tearDown() async throws {
+        Dependencies.shared.remove(for: ChangeTierClient.self)
+        try await Task.sleep(nanoseconds: 20_000_000)
+        XCTAssertNil(sut)
+    }
+
+    func testFetchTiersSuccess() async throws {
+        let changeTierIntentModel: ChangeTierIntentModel = .init(
+            activationDate: Date(),
+            tiers: tiers,
+            currentPremium: .init(amount: "449", currency: "SEK"),
+            currentTier: nil,
+            currentDeductible: nil,
+            canEditTier: true
+        )
+
+        let mockService = MockData.createMockChangeTier(fetchTier: { _ in
+            changeTierIntentModel
+        })
+
+        self.sut = mockService
+
+        let model = ChangeTierViewModel(changeTierInput: .init(source: .changeTier, contractId: "contractId"))
+
+        model.fetchTiers(nil)
+
+        try await Task.sleep(nanoseconds: 30_000_000)
+        assert(model.tiers == tiers)
+        assert(model.tiers.first == tiers.first)
+        assert(model.tiers.count == tiers.count)
```

Comment:
It would be nice to test if other properties are right inside vm. 
Also for other tests